lib/incito-browser/incito.ts

```diff
@@ -648,7 +648,26 @@ function renderView(view, canLazyload) {
     return {tagName, contents, attrs};
 }
 
+export interface IncitoEventMap {
+    started: [];
+    destroyed: [];
+    sectionVisible: [{sectionId: string; sectionPosition: number}];
+    sectionHidden: [{sectionId: string; sectionPosition: number}];
+}
 export default class Incito extends MicroEvent {
+    declare bind: <N extends keyof IncitoEventMap>(
+        name: N,
+        cb: (...args: IncitoEventMap[N]) => void
+    ) => void;
+    declare unbind: <N extends keyof IncitoEventMap>(
+        name: N,
+        cb: (...args: IncitoEventMap[N]) => any
+    ) => void;
+    declare trigger: <N extends keyof IncitoEventMap>(
```

Comment:
I feel like this looks impossible to read/maintain 😅 Can we clean this up/simplify?

Reply:
Roger 👍